.gitlab-ci.yml

```
@@ -402,27 +415,56 @@ publish:disclaimer:
     expire_in: 2w
     paths:
       - disclaimer.txt
-  only:
-    - /^(master|staging|[0-9]+\.[0-9]+\.x)$/
+  rules:
+    - if: $CI_COMMIT_BRANCH =~ /^(master|staging|[0-9]+\.[0-9]+\.x)$/
   tags:
     - mender-qa-worker-generic-light
 
+# excludes non multiplatform build job
 publish:image:
-  after_script:
-    - *export_docker_vars
-    - docker load -i unprivilegedImage.tar
-    - docker tag $DOCKER_BUILD_SERVICE_IMAGE $DOCKER_REPOSITORY:${DOCKER_PUBLISH_TAG}-unprivileged
-    - docker push $DOCKER_REPOSITORY:${DOCKER_PUBLISH_TAG}-unprivileged
+  rules:
+    - when: never
+ 
+# excludes non multiplatform job
+publish:image:saas:
+  rules:
+    - when: never
 
+# excludes non multiplatform job
 publish:image:mender:
-  after_script:
-    - *export_docker_vars
-    - integration_versions=$(release_tool --integration-versions-including $CI_PROJECT_NAME --version $CI_COMMIT_REF_NAME | sed -e 's/origin\///')
-    - docker load -i unprivilegedImage.tar
+  rules:
+    - when: never
+
+publish:image-multiplatform:unprivileged:
+  extends: publish:image-multiplatform
+  variables:
+    GITLAB_REGISTRY_TAG: '${CI_REGISTRY_IMAGE}:${CI_PIPELINE_ID}-unprivileged'
+  script:
+    - regctl image copy ${GITLAB_REGISTRY_TAG} ${DOCKER_REPOSITORY}:${DOCKER_PUBLISH_TAG}-unprivileged
+    - regctl image copy ${GITLAB_REGISTRY_TAG} ${DOCKER_REPOSITORY}:${DOCKER_PUBLISH_COMMIT_TAG}
+    - echo "PUBLISH_IMAGE_DIGEST=${DOCKER_REPOSITORY}@$(regctl image digest ${GITLAB_REGISTRY_TAG})" >> publish.env
+
+publish:image-multiplatform:mender-unprivileged:
+  extends: publish:image-multiplatform:mender
+  variables:
+    GITLAB_REGISTRY_TAG: '${CI_REGISTRY_IMAGE}:${CI_PIPELINE_ID}-unprivileged'
+  script:
+    # Publish the image for all releases
     - for version in $integration_versions; do
-    - docker tag $DOCKER_BUILD_SERVICE_IMAGE $DOCKER_REPOSITORY:mender-${version}-unprivileged
-    - docker push $DOCKER_REPOSITORY:mender-${version}-unprivileged
+    -   regctl image copy ${GITLAB_REGISTRY_TAG} ${DOCKER_REPOSITORY}:mender-${version}-unprivileged
+    -   regctl image copy ${GITLAB_REGISTRY_TAG} ${DOCKER_REPOSITORY}:mender-${version}-unprivileged_${CI_COMMIT_SHA}
     - done
+    - echo "PUBLISH_IMAGE_DIGEST=${DOCKER_REPOSITORY}@$(regctl image digest ${DOCKER_REPOSITORY}:mender-${version}-unprivileged)" >> publish.env
+
+publish:image-multiplatform:saas-unprivileged:
+  extends: publish:image-multiplatform:saas
+  before_script:
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
+    - *export_docker_vars
+    - *docker_login_registries
+    - SOURCE_TAG=staging_${CI_COMMIT_SHA}
+    - DOCKER_PUBLISH_TAG=${CI_COMMIT_REF_NAME}-unprivileged
+    - SERVICE_IMAGE=${DOCKER_REPOSITORY}:${DOCKER_PUBLISH_TAG}
```

```suggestion
    - !reference [publish:image:saas, before_script]
    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
    - *export_docker_vars
    - DOCKER_PUBLISH_TAG=${CI_COMMIT_REF_NAME}-unprivileged
    - SERVICE_IMAGE=${DOCKER_REPOSITORY}:${DOCKER_PUBLISH_TAG}
```
since the `docker_login_registries` anchor is/ anchors are not available from included files... the `!reference` should work though!

---

Nice feature! Didn't know this